### ISSUE TITLE
Project structure cleanup & documentation

### ISSUE THREAD

**Issue by 100ideas** [DIYbiosphere/sphere#195]
- can these be removed from `master` branch until complete:
  - google maps stuff `_layouts/maps.html` & JS code
  - code using `lafeber/world-flags-sprite` - doesn't semantic UI provide flags?
- can `.github` be moved to `docs/github`
- can content files ( `_groups`, `_labs`, `_events` etc) be moved to a subdirectory like `_content/`
  - or could site files (assets, includes etc) be moved to a common `_src` subdirectory

below is a list of TODOs I have added to the code in my branch indicating areas where I am confused: 

`git grep --heading -n --break -C 3 TODO > TODOS.md`

```liquid
_includes/contentMenu.html
1:<!-- TODO tabular menu not being inculded? -->
2-<div class="ui secondary pointing mini tabular grey menu">
3-  <a class="active item" data-tab="overview">Main</a>
4-

_includes/head.html
11-  <link rel="alternate" type="application/rss+xml" title="{{ site.title }}" href="{{ '/feed.xml' | prepend: site.baseurl | prepend: site.url }}">
12-
13-  <!-- Semantic UI -->
14:  <!-- TODO include Source Code Pro font in local assets -->
15-  <link href="https://fonts.googleapis.com/css?family=Source+Code+Pro" rel="stylesheet">
16-  {% css semantic/dist/semantic.min %}
17-
--
26-  {% js vendor/jquery-2.2.2 %}
27-  {% js vendor/jquery.scrollNav.min %}
28-
29:  <!-- TODO warn: injects jquery 3  {% javascript githubComments %} -->
30:  <!-- TODO warn: injects jquery 3  {% javascript githubCommentsEdit %} -->
31-
32-  {% if page.layout == 'landing' %}
33-    <!-- OWL Carousel css (JS transcluded in default.html)-->
--
47-  {% endif %}
48-
49-  <!-- Algolia DocSearch -->
50:  <!-- TODO probably unnecssary; remove? -->
51-  {% if page.layout == 'docs' %}
52:    <!-- TODO include algolia instantsearch.min.js in local assets -->
53-    <link rel="stylesheet" type="text/css" href="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css" />
54-  {% endif %}
55-

_includes/header.html
31-      </div>
32-
33-      <!-- search -->
34:      <!-- TODO we only need one search box on the entries list / landing page -->
35-      <div class="ui right aligned category search item">
36-        <div class="ui blue icon input">
37-          <!-- hook up with js/autocomplete.es6 -->

_includes/keywords.html
2-  <!-- Tags -->
3-  <div>
4-    <div class="ui small labels">
5:      <!-- TODO figure out whats up with these ugly url params -->
6-      {% for tag in page.tags %}
7-      <a class="ui label noul" href="/entries/?q=&idx=diybiosphere&p=0&dFR%5Btags%5D%5B0%5D={{ tag }}">
8-        {{ tag }}

_includes/tabs.html
1:<!-- TODO tabular menu not being inculded? -->
2-<div id="{{ include.id }}" class="ui grid">
3-  <div class="two wide column">
4-    <div class="ui small vertical fluid tabular menu">

_layouts/default.html
17-    <!-- Google Analytics -->
18-    <script src="https://www.google.com/jsapi"></script>
19-
20:    <!-- Semantic UI js? TODO broken-->
21-    {% javascript semantic/dist/semantic.min.js %}
22-    <script type="text/javascript">$('.ui.dropdown').dropdown();</script>
23-    <script type="text/javascript">$('.tabular.menu .item').tab();</script>
24-
25:    <!-- TODO figure out what's going on here... already included via main.scss? -->
26-      <!-- https://github.com/lafeber/world-flags-sprite -->
27-      <!-- <link rel="stylesheet" type="text/css" href="//cloud.github.com/downloads/lafeber/world-flags-sprite/flags16.css" /> -->
28-
--
42-      {% javascript carousel %}
43-      <script type="text/javascript">$('.message .close').on('click', function() {$(this).closest('.message').transition('fade');});</script>
44-
45:      <!-- TODO depends on _plugins/dataDoc.rb - CURRENTLY COMMENTED OUT -->
46-      <!-- {% javascript googleMap %} -->
47:      <!-- TODO Algolia search for #index-stats-container on index.html#90 - CURRENTLY COMMENTED OUT -->
48-      <!-- {% javascript lastEdited_instantsearch %} -->
49-    {% endif %}
50-
--
55-    {% endif %}
56-
57-    {% if page.layout == 'docs' %}
58:    <!-- TODO already included in head.html -->
59-      <script type="text/javascript" src="https://cdn.jsdelivr.net/docsearch.js/2/docsearch.min.js"></script>
60-      <script type="text/javascript">$('.menu .item').tab();</script>
61-      <script type="text/javascript">$('.ui.accordion').accordion({onOpen: function (item) {setCookie('acordionIndex',this.index);}}).accordion('open', getCookie('acordionIndex'));</script>

_layouts/entries.html
4-
5-<div class="ui grid container xo padding top twofold">
6-
7:  <!-- TODO make tall & skinny -->
8-  <!-- FILTERS -->
9-  <div class="four wide column">
10-    <div class="ui segments">

assets/js/googleMap.js
21-  var noResultsTemplate = Hogan.compile($('#no-results-template').text());
22-
23-  // Map initialization
24:  // TODO depends on _plugins/dataDoc.rb - CURRENTLY COMMENTED OUT
25-  var map = new google.maps.Map($map.get(0), {
26-    streetViewControl: false,
27-    zoom: 4,

assets/js/vendor/owl.carousel.js
326-			var clones = [],
327-				items = this._items,
328-				settings = this.settings,
329:				// TODO: Should be computed from number of min width items in stage
330-				view = Math.max(settings.items * 2, 4),
331-				size = Math.ceil(items.length / 2) * 2,
332-				repeat = settings.loop && items.length ? settings.rewind ? view : Math.max(view, size) : 0,


index.html
18-          <h1 class="ui  header"><div class="ui huge inverted header">DIYbiosphere</div></h1>
19-          <div class="ui huge inverted header">An open source project to collect DIYbio initiatives from all over the world. <br> Search, browse, update, or add an initiative!</div>
20-          <div class="ui medium inverted header">
21:            <!-- TODO what is up with the following google.ch click-through link to github? -->
22-            To contribute all you need is a <a href="https://www.google.ch/url?sa=t&rct=j&q=&esrc=s&source=web&cd=1&cad=rja&uact=8&ved=0ahUKEwig3Zb8wMTVAhUPmbQKHU31DbwQFggoMAA&url=https%3A%2F%2Fgithub.com%2Fjoin&usg=AFQjCNF6nezHQWX1hKwEFQVYRrUheS9_Ig">GitHub account!</a>
23-          </div>
24-          <a href="/about/diybiosphere"><button class="ui inverted big button">LEARN MORE</button></a>
```

**Comment by sabgaby** [comment]
- We could move all files: [CONTRIBUTING](https://help.github.com/articles/setting-guidelines-for-repository-contributors/), [CODE_OF_CONDUCT](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/), [SUPPORT](https://help.github.com/articles/adding-support-resources-to-your-project/) (we don't have it for now), [ISSUE_TEMPLATE](https://help.github.com/articles/creating-an-issue-template-for-your-repository/) and [PULL_REQUEST_TEMPLATE](https://help.github.com/articles/creating-a-pull-request-template-for-your-repository/) inside `docs` folder. But we would have to leave the [LICENSE](https://help.github.com/articles/adding-a-license-to-a-repository/) file in the root folder.
- the collection folders for now will have to be in the root directory, but in Jekyll 3.7 we will be able to specify a `collections_dir:` in the config file. See [Gathering your collections](https://jekyllrb.com/docs/collections/)

**Comment by 100ideas** [comment]
Thanks @sabgaby, sounds good. I will leave LICENSE in root and move the rest... and update the `cibuild` script as appropriate?

**Comment by 100ideas** [comment]
I finished reviewing the major files in the codebase. I wanted to understand the JS assets in particular to reduce the chances of getting seriously confused when working on the `entries` layout and associated `includes` like those for the scrollNav and instantsearch.

Could you take a quick look at the **bold** files and let me know if any of them are critical? We can work on cleaning things up later, in the meantime I will make sure to preserve any that we know are important.

### unused and/or mystery files

- **bold** - can't find them included or used anywhere in the codebase, recommend removing from `master` branch or providing more documentation
- *italic* - unsure what these do, although they are referenced. need documenation.

.
├── *DataReusePlan.txt*
├── _data
│   └── **images.yml**
├── _includes
│   ├── **accordion.html**
│   ├── *githubComments.html*
│   ├── *githubCommentsEdit.html*
│   ├── **map.html**
│   ├── **pills.html**
├── _layouts
│   ├── **maps.html**
│   └── *post.html* (defined as default layout in _config.yml?)
├── _plugins
│   ├── **algolia_push.rb**
│   ├── **dataDoc.rb**
│   ├── **jekyll-geocode.rb**
├── assets
│   ├── js
│   │   ├── **form_places.es6**
│   │   ├── *githubComments.js*
│   │   ├── *githubCommentsEdit.js*
│   │   ├── **githubReactions.js**
│   │   ├── *googleMap.js*
│   │   ├── **heroInstantsearch.es6**
│   │   ├── **lastEdited_instantsearch.es6** (target div commented out in index.html...)
│   │   ├── **main.es6**
│   │   ├── **mapInstantSearch.es6**
│   │   ├── **maps.es6**
│   │   ├── **quickSearch_places.es6**
│   │   └── vendor
│   │       ├── **brands.js**
│   │       ├── **jquery.scrollNav.min.js**
│   │       ├── **owl.carousel.js**
│   │       ├── *regular.js*
│   │       ├── *scrollNav.js*
│   │       └── *solid.js* (from fontAwesome?)
├── docs
│   ├── **drafts
│   │   └── whatever.md**
└── pages
    └── **maps.html**
    
---

### more-complete file listing

.
├── 404.html
├── **DataReusePlan.txt**
├── Gemfile
├── Gemfile.lock
├── README.md
├── _config.yml
├── _data
│   ├── database.csv
│   ├── database.json
│   └── **images.yml**
├── _events
│   ├── BioDesignChallenge
│   ├── BioFabbing
│   ├── BioHTP
│   ├── GlobalCommunityBioSummit
│   └── Igem
├── _groups
│   ├── BeInToBiohacking
│   ├── BioHackSyd
│   ├── BioHubIl
│   ├── BioTinkeringBerlin
│   ├── Biogarage
│   ├── BiohackersNYC
│   ├── BiohackingMexico
│   ├── CapCityBiohackers
│   ├── ChicagoBiohacking
│   ├── DIYNeurotech
│   ├── DIYbioAuckland
│   ├── DIYbioBuenosAires
│   ├── DIYbioChicago
│   ├── DIYbioGroningen
│   ├── DIYbioIreland
│   ├── DIYbioIsrael
│   ├── DIYbioMexico
│   ├── DIYbioPerth
│   ├── DIYbioSingapore
│   ├── GeneGarage
│   ├── Hackstronauts
│   ├── MNDIYbio
│   ├── ManilaBiopunk
│   └── TriDIYbio
├── _includes
│   ├── label/...
│   ├── messages/...
│   ├── other/...
│   ├── subtabs/...
│   ├── tabs/...
│   ├── **accordion.html**
│   ├── cc-image.html
│   ├── contentMenu.html
│   ├── contentTab.html
│   ├── editOnGitHub.html
│   ├── footer.html
│   ├── *githubComments.html*
│   ├── *githubCommentsEdit.html*
│   ├── head.html
│   ├── header.html
│   ├── info-box.html
│   ├── keywords.html
│   ├── **map.html**
│   ├── **pills.html**
│   ├── promotions.html
│   ├── sidebar.html
│   ├── social-icons.html
│   ├── tabs.html
│   └── video.html
├── _incubators
│   ├── BerkleyBiolabs
│   ├── BioTechAndBeyond
│   ├── BlueCity
│   ├── HarlemBiospace
│   ├── Hax
│   ├── IndieBio
│   └── RebelBio
├── _labs
│   ├── ArtScienceBlr
│   ├── BioArtLaboratories
│   ├── BioDidact
│   ├── BioFoundry
│   ├── BioMakersLabPeru
│   ├── BioNyfiken
│   ├── BioQuisitive
│   ├── BioTehna
│   ├── BioTown
│   ├── Biocurious
│   ├── BiologiGaragen
│   ├── BiologikLabs
│   ├── Bioscope
│   ├── BiotechWithoutBorders
│   ├── BosLab
│   ├── BricoBio
│   ├── BrmBiolab
│   ├── Bugss
│   ├── CapitalAreaBioSpace
│   ├── ChiTownBio
│   ├── CounterCultureLabs
│   ├── DIYBioBCN
│   ├── DIYBioTech
│   ├── DIYbioBelgium
│   ├── DIYbioToronto
│   ├── DenverBiolabs
│   ├── FLab
│   ├── GaroaOpenBiolab
│   ├── GaudiLabs
│   ├── Genspace
│   ├── Hackuarium
│   ├── HiveBio
│   ├── IndieLab
│   ├── LEprouvette
│   ├── LaJollaBiolab
│   ├── LaPaillase
│   ├── LaPaillaseSaone
│   ├── LeBiomeHackLab
│   ├── LifePatch
│   ├── LondonBiohackerspace
│   ├── MadLabBiolab
│   ├── OpenBioLabGraz
│   ├── OpenBioLabs
│   ├── OpenScienceNet
│   ├── OpenWetlab
│   ├── PechBlenda
│   ├── Reagent
│   ├── Symbiolab
│   ├── TheLAb
│   ├── TheWetlab
│   └── TrySciBiolab
├── _layouts
│   ├── default.html
│   ├── docs.html
│   ├── entries.html
│   ├── entry-example.html
│   ├── entry.html
│   ├── form.html
│   ├── **image-info.html**
│   ├── landing.html
│   ├── **maps.html**
│   ├── page.html
│   └── *post.html* (defined as default layout in _config.yml)
├── _networks
│   ├── BioChanges
│   ├── BioDesignFTRW
│   ├── DIYbio.org
│   ├── Hackteria
│   ├── NYCBio
│   └── SyntechBio
├── _plugins
│   ├── **algolia_push.rb**
│   ├── **dataDoc.rb**
│   ├── determine-collection.rb
│   ├── ext.rb
│   ├── include_absolute.rb
│   ├── **jekyll-geocode.rb**
│   ├── jekyll-sibling-images.rb
│   ├── references.rb
│   └── timestamp.rb
├── _projects
│   ├── BioDisplay
│   ├── DIYbiosphere
│   ├── KombuchaGenomics
│   ├── Metafluidics
│   ├── OYEP
│   ├── OpenGenX
│   ├── OpenInsulin
│   ├── OpenPcr
│   ├── RealVeganCheese
│   └── RoninGenetics
├── _references.md
├── _site
│   ├── 404.html
│   ├── about
│   ├── assets
│   ├── docs
│   ├── entries
│   ├── events
│   ├── favicon.ico
│   ├── feed.xml
│   ├── form
│   ├── groups
│   ├── incubators
│   ├── index.html
│   ├── labs
│   ├── **maps**
│   ├── networks
│   ├── projects
│   ├── robots.txt
│   ├── sitemap.xml
│   └── startups
├── _startups
│   ├── AminoLabs
│   ├── ChaiBio
│   ├── OpenTrons
│   ├── RotterZwam
│   └── TheODIN
├── assets
│   ├── css
│   │   ├── base.scss
│   │   ├── main.scss
│   │   ├── modules
│   │   │   ├── autocomplete.scss
│   │   │   ├── brands-colors.scss
│   │   │   ├── clipboard.scss
│   │   │   ├── docsearch.scss
│   │   │   ├── *sidebar-navscroll.scss*
│   │   │   ├── syntax.scss
│   │   │   └── utils.scss
│   │   ├── variables.scss
│   │   └── vendor
│   │       ├── cc-icons.css
│   │       ├── flags-16.scss
│   │       ├── **gh-code.scss**
│   │       ├── **gitter-sidecar.scss**
│   │       ├── octicons.css
│   │       ├── owl.carousel.css
│   │       ├── owl.carousel.min.css
│   │       ├── owl.theme.default.css
│   │       └── owl.theme.default.min.css
│   ├── gifs/...
│   ├── img/...
│   ├── js
│   │   ├── autocomplete.es6
│   │   ├── browse_instantsearch.es6
│   │   ├── carousel.js
│   │   ├── docs.es6
│   │   ├── **form_places.es6**
│   │   ├── *githubComments.js*
│   │   ├── *githubCommentsEdit.js*
│   │   ├── **githubReactions.js**
│   │   ├── *googleMap.js*
│   │   ├── **heroInstantsearch.es6**
│   │   ├── **lastEdited_instantsearch.es6** (target div commented out in index.html...)
│   │   ├── **main.es6**
│   │   ├── **mapInstantSearch.es6**
│   │   ├── **maps.es6**
│   │   ├── **quickSearch_places.es6**
│   │   └── vendor
│   │       ├── algolia-algoliasearch_helper-2.js
│   │       ├── algolia-autocomplete-0.29.0.js
│   │       ├── algoliasearch-3.24.8.js
│   │       ├── **brands.js**
│   │       ├── clipboard-1.7.1.js
│   │       ├── fontawesome.js
│   │       ├── hogan-3.0.2.js
│   │       ├── instantsearch.js
│   │       ├── jquery-2.2.2.js
│   │       ├── **jquery.scrollNav.min.js**
│   │       ├── owl-carousel-2.2.1.js
│   │       ├── **owl.carousel.js**
│   │       ├── *regular.js*
│   │       ├── *scrollNav.js*
│   │       └── *solid.js* (from fontAwesome?)
│   ├── *semantic/...* (just include semantic.min.js?)
│   ├── svg
│   └── videos
├── docs
│   ├── EntryTemplate.md
│   ├── about
│   │   ├── code-of-conduct.md
│   │   ├── community.md
│   │   ├── contact.md
│   │   ├── contributor-terms.md
│   │   ├── diybiosphere.md
│   │   ├── faq.md
│   │   ├── philosophy.md
│   │   ├── support-us.md
│   │   └── terms-of-use.md
│   ├── basics
│   │   ├── design.md
│   │   ├── documentation.md
│   │   ├── entries.md
│   │   └── software.md
│   ├── development.md
│   ├── **drafts
│   │   ├── draft2.md
│   │   ├── mapdraft.md
│   │   └── whatever.md**
│   ├── *github (move to here ok?)
│   │   ├── CHANGELOG.md
│   │   ├── CODE_OF_CONDUCT.md
│   │   ├── CONTRIBUTING.md
│   │   ├── ISSUE_TEMPLATE.md
│   │   ├── LICENSE.md
│   │   ├── PULL_REQUEST_TEMPLATE.md
│   │   └── ROADMAP.md*
│   ├── guides
│   │   ├── contributing.md
│   │   ├── github-eli5.md
│   │   ├── glossary.md
│   │   ├── markdown.md
│   │   ├── permissions.md
│   │   ├── support.md
│   │   ├── workflow.md
│   │   └── writing.md
│   ├── introduction
│   │   ├── getting-started.md
│   │   ├── introduction.md
│   │   └── overview.md
│   ├── tutorials
│   │   ├── AvocadoLab
│   │   │   ├── AvocadoLab-header.png
│   │   │   ├── AvocadoLab-logo.png
│   │   │   ├── AvocadoLab.md
│   │   │   ├── EntryExample.png
│   │   │   ├── EntryExampleAnatomy.png
│   │   │   ├── RenderedExample.md
│   │   │   ├── image-example.jpg
│   │   │   └── sidebyside.png
│   │   ├── add-entry
│   │   │   ├── members.md
│   │   │   └── users.md
│   │   ├── edit-entry
│   │   │   ├── members.md
│   │   │   └── users.md
│   │   ├── *form
│   │   │   ├── form-save.md
│   │   │   ├── form-upload-members.md
│   │   │   ├── form-upload-teams.md
│   │   │   ├── form-upload.md
│   │   │   └── form-write.md*
│   │   ├── local-install
│   │   │   ├── linux.md
│   │   │   ├── macos.md
│   │   │   └── windows.md
│   │   ├── noting-copyright.md
│   │   ├── tutorials.md
│   │   ├── using-atom.md
│   │   ├── using-github.md
│   │   └── using-githubdesktop.md
│   └── welcome.md
├── favicon.ico
├── feed.xml
├── index.html
├── pages
│   ├── database.html
│   └── **maps.html**
└── script
    └── cibuild

**Comment by 100ideas** [comment]
closing since it's so old - will revisit
